### ISSUE TITLE
OpenAPI Validation for CR

### ISSUE THREAD

**Issue by AdheipSingh** [druid-io/druid-operator#19]
Since the yaml is large lot of parameters are sometimes go unoticed. Can we add a validation openapi for custom resource so we can be checked if a REQUIRED parameter is missing.

**Comment by himanshug** [comment]
it might be simpler to just add code verifying the provided spec . I imagine openapi would require modifying the generated  `deploy/crds/druid.apache.org_druids_crd.yaml`  file . operator-sdk is capable of generating openapi things  which I had tried earlier and that generated a lot more  than I  was comfortable with.
E.g. I tried again today, and here is the diff produced for `deploy/crds/druid.apache.org_druids_crd.yaml` : https://gist.github.com/himanshug/b570b05b8ec4b150e626df5f1ca0a5f3

that aside, it is nice to add validations whether with openapi or with custom code. 
I guess main advantage of openapi based validation would be that k8s would itself reject cr if it wasn't valid.
